src/search.cpp

```diff
@@ -1250,6 +1250,11 @@ namespace {
       {
           Depth r = reduction<PvNode>(improving, depth, moveCount);
 
+#ifdef ANTI
+          if (pos.is_anti() && pos.can_capture())
+              r -= r ? ONE_PLY : DEPTH_ZERO;
+          else
```

Comment:
Sorry for the overly long message. I would like to draw some useful conclusions from the way this discussion evolved. Take your time to read, reflect, and reply.

----------------------------------

> FYI (looking forward), I don't think I'd reject a PR without first discussing reasons (such as criteria) to consider rejecting it (hopefully historically I've been consistent in this; if not I should become consistent with it in the future).

Sounds good.

> So here are the specific quotes where new (and IMHO relevant) conversation topics were introduced:

> > I am just a bit confused about the state of this PR. Is it accepted, rejected, or deferred until concerns about functionality/performance are addressed? I see neither of these, but just some ideas for further improvements (which are interesting, but the evaluation of the PR should not depend on those results unless they reveal a bug or the like).

I think that asking for the status of the PR is on topic, but I guess I could have maybe kept it more simple.

>> Fearing not to be able to improve further should not be a reason to keep you from improving, and by the way, why it is important that you improve over it with a similar patch?

This was just meant to point out the illogicality, but I see that this might provoke a more general discussion that was not intended.

---------------------------------------------------

I calmly went over the thread again and tried to condense my criticism. My conclusion is that from my point of view you mentioned a lot of rather diffuse reasons why you are critical of the PR. This way I did not know what really are your concerns regarding the PR that have to be addressed and it made me think that there might be a reason you did not mention and that you are instead collecting arguments. I am still not entirely clear about your main motivation, but I think that you were mainly concerned that the change would make the LMR code and the search in general more sensitive to changes. This is not directly measurable, but it could be estimated by applying a few (random) changes to the search and for each measure and compare the Elo difference for the cases that the branches are or are not rebased on the PR. My interpretation of your concerns and the proposed solution are mainly meant as an example of how we could very quickly come up with interesting solutions if we both focus more on constructive feedback.

So what I would like you to do in the future is that if your feedback is not very concrete to reflect more about what you are in essence really concerned about even if that means that I have to wait longer for a response (if it takes long, you could e.g. simply leave a quick comment to let me know) and then condense your feedback so that we can more constructively address your concern(s).